Python/ceval_macros.h

```diff
@@ -387,7 +387,34 @@ stack_pointer = _PyFrame_GetStackPointer(frame);
 
 /* Tier-switching macros. */
 
-#define GOTO_TIER_TWO() goto enter_tier_two;
+#ifdef _Py_JIT
+#define GOTO_TIER_TWO(EXECUTOR)                            \
+    do {                                                   \
+        jit_func jitted = (EXECUTOR)->jit_code;            \
+        next_instr = jitted(frame, stack_pointer, tstate); \
+        Py_DECREF(tstate->previous_executor);              \
+        tstate->previous_executor = NULL;                  \
+        frame = tstate->current_frame;                     \
+        if (next_instr == NULL) {                          \
+            goto resume_with_error;                        \
+        }                                                  \
+        stack_pointer = _PyFrame_GetStackPointer(frame);   \
+        DISPATCH();                                        \
+    } while (0)
+#else
+#define GOTO_TIER_TWO(EXECUTOR) \
+do { \
+    next_uop = (EXECUTOR)->trace; \
+    assert(next_uop->opcode == _START_EXECUTOR || next_uop->opcode == _COLD_EXIT); \
+    goto enter_tier_two; \
+} while (0)
```

Comment:
Should probably keep the same indentation as the other side of the `#ifdef` (maybe add a `do`/`while` to `GOTO_TIER_ONE` too).

```suggestion
#define GOTO_TIER_TWO(EXECUTOR)                                                        \
    do {                                                                               \
        next_uop = (EXECUTOR)->trace;                                                  \
        assert(next_uop->opcode == _START_EXECUTOR || next_uop->opcode == _COLD_EXIT); \
        goto enter_tier_two;                                                           \
    } while (0)
```